test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -35,6 +35,34 @@ import (
 )
 
 var _ = Describe("Docker", func() {
+	Describe("Cluster Deletion", func() {
```

Comment:
I would expect that we do these assertions for each of the Cluster creation contexts below instead of as a separate test.